src/auth/jwt.strategy.ts

```diff
@@ -0,0 +1,44 @@
+import { Strategy } from 'passport-jwt';
+import { PassportStrategy } from '@nestjs/passport';
+import { Injectable, UnauthorizedException } from '@nestjs/common';
+import { Student } from '../student/student.entity';
+
+// co bedziemy przechowywać, token zalogowania
+export interface JwtPayload {
+   id: string;
+}
+
+// wyciaga ciatko z req, jezeli istnieje zwracamy, nie to null
+function cookieExtractor(req: any): null | string {
+   return req && req.cookies ? req.cookies?.jwt ?? null : null;
+}
+
+export const jwtKey =
+   'nhoiasdhfoidsh9873289rfw7dsgfe023@RAWESFdsddR32qwrfEDggraDer32h 88 h98qwye79t3 rewr sdfsdf 3 232reF#@4 23r$#@4';
+
+@Injectable()
+export class JwtStrategy extends PassportStrategy(Strategy) {
+   constructor() {
+      super({
+         jwtFromRequest: cookieExtractor,
+         secretOrKey: jwtKey,
+      });
+   }
+
+   async validate(payload: JwtPayload, done: (error, user) => void) {
+      if (!payload || !payload.id) {
+         return done(new UnauthorizedException(), false);
+      }
+
+      // sprawdzanie typu uzytkownika, dodatkowy parametr w metodzie i endpoincie np. login/student, login/hr?
+      // Poki co dla student
+      const user = await Student.findOneBy({
+         currentSessionTokenId: payload.id,
+      });
```

Comment:
Here we will take User table and there will be role used to redirect user to proper validated routes